tests/callbacks_test.cpp

```diff
@@ -562,3 +562,66 @@ TEST_CASE("TimerStopCallbackTest", "[CallbacksTest]")
   // Add some time to account for the function to return.
   REQUIRE(timer.toc() < 2);
 }
+
+/**
+ * Make sure the ProgressBar callback will show the progress on the specified
+ * output stream if the MaxIterations parameter of the optimizer is 0.
+ */
+TEST_CASE("ProgressBarCallbackNoMaxIterationsTest", "[CallbacksTest]")
+{
+  SGDTestFunction f;
+  arma::mat coordinates = f.GetInitialPoint();
+
+  StandardSGD s(0.0003, 1, 0, 1e-9, true);
```

Comment:
If we set `MaxIterations() = 0` we should use a higher threshold, to decrease the test runtime, looks like with that change we added about 20 minutes to the test/build time.

Reply:
Hi, @zoq I increased the `tolerance` value to `DBL_MAX` should I increase the step size too? Thanks.